packages/firestore/src/util/async_queue.ts

```diff
@@ -337,8 +338,19 @@ export class AsyncQueue {
           deferred.resolve();
           this.backoff.reset();
         } catch (e) {
-          logDebug(LOG_TAG, 'Retryable operation failed: ' + e.message);
-          this.backoff.backoffAndRun(retryingOp);
+          // We only retry errors that match the ones thrown by IndexedDB (see
+          // https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction/error)
+          if (
+            (typeof DOMException !== 'undefined' &&
```

Comment:
`DOMException` can mean just about any failure in the browser. A better way to handle this would be to take the [`transaction.error` property](https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/simple_db.ts#L438) and wrap it in our own `PersistenceError` error type (or similar). That way it's unambiguous that it's an error that came from IndexedDB. This also sidesteps the issue of the availability of the `DOMException` type.

Reply:
Done. I am now wrapping exceptions thrown by `abort` and `error` in their own types.